rust/kernel/src/sync.rs

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: GPL-2.0
+
+use core::cell::UnsafeCell;
+use core::ops::{Deref, DerefMut};
+use core::fmt;
+
+use crate::bindings;
+
+#[cfg(CONFIG_DEBUG_SPINLOCK)]
+compile_error!("CONFIG_DEBUG_SPINLOCK not yet implemented");
+
+#[cfg(CONFIG_DEBUG_MUTEX)]
+compile_error!("CONFIG_DEBUG_MUTEX not yet implemented");
+
+pub struct Mutex<T> {
+    lock: bindings::mutex,
+    data: UnsafeCell<T>,
+}
+
+impl<T> Mutex<T> {
+    /// Create a new Mutex
+    pub fn new(data: T, name: &'static str) -> Self {
+
+        let lock = bindings::mutex::default();
+        // TODO: debug foo
```

Comment:
?

Reply:
there are special debug features behind CONFIG_DEBUG_MUTEX, which are not yet implemented

Reply:
Please expand the comment then :-)

Reply:
Still applicable.

Reply:
like this?
~~~
// TODO: write mutex debug traces like .magic

Reply:
Yes, that'd be good. TODO should also mention the unused `name` param.

Reply:
pushed it.